contracts/VaultV3.vy

```diff
@@ -786,14 +797,20 @@ def _redeem(
             # We update the previous_balance variable here to save gas in next iteration.
             previous_balance = post_balance
 
-            # Reduce what we still need.
+            # Reduce what we still need. Safe to use assets_to_withdraw 
+            # here since it has been checked agains requested_assets
             assets_needed -= assets_to_withdraw
 
         # If we exhaust the queue and still have insufficient total idle, revert.
         assert curr_total_idle >= requested_assets, "insufficient assets in vault"
         # Commit memory to storage.
         self.total_debt = curr_total_debt
 
+    # Check if there is a loss and a non-default value was set.
+    if assets > requested_assets and max_loss < MAX_BPS:
+        # The loss is withen the allowed range.
+        assert assets - requested_assets <= assets * max_loss / MAX_BPS, "to much loss"
```

Comment:
maybe a `assert max_loss <= 10_000` sanity check here could be helpful.
weird edge case if someone puts max_loss to an extremely high number (e.g. 2**256-1) it could overflow and get unexpected result?

Reply:
Good point, it could be anything. but the  "and max_loss < MAX_BPS" should take care of weird inputs right? Since that needs to be true for max_loss to ever get used